### DISCUSSION TITLE
How publishing to firefox works

### DISCUSSION THREAD

**Discussion by davidabou** [wxt-dev/wxt#575] [Q&A] [answered]
Hello everybody, I have made my extension with WXT and I want to publish it to firefox but, the online documentation of this isn't very clear. With the command `wxt build -b firefox` I dont know what is it for.

I have already zip my project with `wxt zip -b firefox` but because the zip is compiled firefox want a file that explain step by step to create a copy of my code.

Someone can explain me how to publish ?

**Comment by aklinker1** [comment, marked as answer]
1. Make sure your extension works in Firefox. WXT creates extensions for Firefox using MV2. To build your extension for Firefox, type `wxt build -b firefox`. This makes a folder .output/firefox-mv2. You can test your extension in Firefox by following [these instructions](https://extensionworkshop.com/documentation/develop/temporary-installation-in-firefox/).

2. After testing, create two ZIP files with the command `wxt zip -b firefox`. You'll get `.output/*-firefox.zip` and `.output/*-sources.zip`. These are the files you need to [submit your extension on the Firefox addon store](https://extensionworkshop.com/documentation/publish/submitting-an-add-on/).
   - Upload `*-firefox.zip` first. This is your extension ready to use.
   - Then under the "Do you need to submit source code" section, select yes and upload `*-sources.zip`.

**Reply by davidabou** [reply]
Thanks !
